### PR TITLE
Support Parsing `ALTER PACKAGE` in openGauss

### PR DESCRIPTION
**Pull request by everly-gif**
Ref #17736 

Changes proposed in this pull request:
- Adds `ALTER PACKAGE` GRAMMAR
- Supports Parsing `ALTER PACKAGE` in openGauss
- Adds tests

✅ Builds locally

### CODE REVIEW

**Review by strongduanmu — CHANGES_REQUESTED** [review]
Hi @everly-gif, can you fix ci error first?

**Comment by everly-gif** [comment]
Hi error is being caused by other merged PRs, I have synced my branch with latest changes, will update when it clears.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/Keyword.g4 [hunk 1]
```diff
@@ -596,6 +596,9 @@ EXCEPT
     : E X C E P T
     ;
 
+PACKAGE
+    : P A C K A G E
+    ;
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/Keyword.g4:601]: Please add a new line after `;`

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Review by strongduanmu — APPROVED** [review]
Good job, mergd.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/Keyword.g4 (hunk 1 above)
Comment (everly-gif) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-opengauss/src/main/antlr4/imports/opengauss/Keyword.g4:601]: done

**Pull request merged**